shared/src/service/transport.ts

```diff
@@ -172,5 +183,10 @@ export class ClientTransportService implements TransportService {
         this.authenticatedPromise = undefined;
       }
     }
+
+    if (this._accessToken) {
```

Comment:
As I suggested in Chat, I would call this in the `submit` method right before the call to `gapi.client.request` so you can be sure that you use the most recently provided accessToken value.

Reply:
Ahh my apologies. With the way the diff viewer collapsed the common code, I couldn't see that this was calling `gapi.client.setToken` irregardless if any of the other conditional blocks were true. This accomplishes what I was mentioning in the previous comment.